test.sh

```diff
@@ -39,6 +55,8 @@ set -ex
 go test -count=1 "$@" ./...
 
 # Configure linters in .golangci.yml
-go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.50.1
-./bin/golangci-lint run
-
+expected_version=v1.50.1
+if ! golangci-lint version | grep -w "$expected_version" >/dev/null 2>&1; then
```

Comment:
No need for all this complexity. If the right version is already installed, `go install` is very fast.